### PR TITLE
[sdk/v2] Fill out RawConfig.

### LINKED ISSUE #548: Ensure bridge is aware of GetRawConfig
Without this, we can get a panic where the provider is using that code - GetRawConfig returns a cty.Value

https://github.com/pulumi/pulumi-auth0/issues/145
https://github.com/pulumi/pulumi-aws/issues/2017

### PR DESCRIPTION
**Pull request by pgavlin**
These changes update the sdk/v2 shim to fill out the RawConfig field on
InstanceState and InstanceDiff. TF providers may inspect the value of
this field to determine the differences between an attribute that has
a default value because it is was not specified and an attribute that
has been explicitly set to its default value.

Fixes #548.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -208,4 +208,4 @@ require (
 	sourcegraph.com/sourcegraph/appdash v0.0.0-20211028080628-e2786a622600 // indirect
 )
 
-replace github.com/hashicorp/terraform-plugin-sdk/v2 => github.com/pulumi/terraform-plugin-sdk/v2 v2.0.0-20220505215311-795430389fa7
+replace github.com/hashicorp/terraform-plugin-sdk/v2 => ../terraform-plugin-sdk
```
Comment (stack72) [inline go.mod:211]: I think this replace has to change back right?

File: pkg/tfshim/sdk-v2/cty.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+	"github.com/hashicorp/go-cty/cty"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+// makeResourceRawConfig converts the decoded Go values in a terraform.ResourceConfig into a cty.Value that is
+// appropriate for Instance{Diff,State}.RawConfig.
+func makeResourceRawConfig(config *terraform.ResourceConfig, resource *schema.Resource) cty.Value {
+	original := schema.HCL2ValueFromConfigValue(config.Raw)
+	coerced, err := resource.CoreConfigSchema().CoerceValue(original)
```
Comment (stack72) [inline pkg/tfshim/sdk-v2/cty.go:14]: Ok so this was what I was missing!!!

**Review by stack72 — APPROVED** [review]
1 small think about go.mod

File: pkg/tfshim/sdk-v2/cty.go (hunk 2 above)
Comment (pgavlin) [reply pkg/tfshim/sdk-v2/cty.go:14]: Yeah, it took my distressingly long to find that. I implemented the equivalent twice before noticing it.

File: go.mod (hunk 1 above)
Comment (pgavlin) [reply go.mod]: Yeah, just fixed that

**Comment by pgavlin** [comment]
Downstream tests are failing due to the need for those modules to reference the updated terraform-plugin-sdk fork.

**Pull request merged**